### ISSUE TITLE
Allow int varref with other numeric operands in a multiplicative expression does not work with singletons

### ISSUE THREAD

**Issue by rdulmina** [ballerina-platform/ballerina-lang#36074]
**Description:**
With https://github.com/ballerina-platform/ballerina-lang/pull/35616 we have allowed  `int*float, float*int, int* decimal, decimal*int, float/int, decimal/int, float%int` and `decimal%int`. in multiplicative expression. But it is not working with singleton types. eg:

```bal
5 a = 5;
float b = 2;
float e = a * b; // gives error operator '*' not defined for '5' and 'float'
```

**Comment by chiranSachintha** [comment]
[spec](https://ballerina.io/spec/lang/2021R1/#multiplicative-expr) says,
```
The static type of both operand expressions is required to be a subtype of the same basic type; 
this basic type will be the static type of the result. The following basic types are allowed: 
```
So we need to disallow the following kinds of scenarios. right?
```
int*float, float*int, int* decimal, decimal*int,
```

**Comment by rdulmina** [comment]
> [spec](https://ballerina.io/spec/lang/2021R1/#multiplicative-expr) says,
> 
> ```
> The static type of both operand expressions is required to be a subtype of the same basic type; 
> this basic type will be the static type of the result. The following basic types are allowed: 
> ```
> 
> So we need to disallow the following kinds of scenarios. right?
> 
> ```
> int*float, float*int, int* decimal, decimal*int,
> ```
@chiranSachintha this is not in the spec yet. See https://github.com/ballerina-platform/ballerina-spec/issues/261

**Comment by rdulmina** [comment]
Fixed with #36074
